File: library/src/main/java/com/example/library/dto/response/SyncDataResponse.java
package com.example.library.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;


@Data
@AllArgsConstructor
public class SyncDataResponse {

    private List<BookSyncResponse> books;

    private List<CategorySyncResponse> categories;

    private List<MemberSyncResponse> members;

    private List<LoanSlipSyncResponse> loanSlips;

    private LocalDateTime timeSync;
}

File: MyApplication/app/src/main/java/com/example/myapplication/data/remote/dto/response/SyncDataResponse.kt
package com.example.myapplication.data.remote.dto.response

import com.example.myapplication.data.local.entity.BookEntity
import com.example.myapplication.data.local.entity.CategoryEntity
import com.example.myapplication.data.local.entity.LoanSlipEntity
import com.example.myapplication.data.local.entity.MemberEntity
import java.time.LocalDateTime
import java.time.LocalTime

data class SyncDataResponse(
    val books : List<BookSyncResponse>,
    val categories : List<CategorySyncResponse>,
    val members : List<MemberSyncResponse>,
    val loanSlips : List<LoanSlipSyncResponse>,
    val timeSync : LocalDateTime
)